src/components/ImageSelector.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { ReactElement, useState } from 'react';
+
+import AmazonS3Service from '../services/AmazonS3Service';
+
+export default function ImageSelector(): ReactElement {
+  const [image, setImage] = useState({ imageBinary: '', imageType: '' });
+
+  return (
+    <>
+      {/* Displays the image selector if no image is selected */}
+      {!image.imageBinary && (
+        <input
+          type="file"
+          accept="image/jpeg, image/png, image/gif"
+          onChange={(event) => {
+            AmazonS3Service.onFileChange(event)
+              .then((image) => setImage(image))
+              .catch((err) => alert(err));
+          }}
+        />
+      )}
+      {/* Displays the image if an image is selected */}
+      {image.imageBinary && <img src={image.imageBinary} alt="" />}
+      {/* Displays the remove image button if an image is selected */}
+      {image.imageBinary && (
+        <button onClick={() => setImage({ imageBinary: '', imageType: '' })}>Remove image</button>
+      )}
+      {/* Displays the upload image button if an image is selected */}
+      {/* TODO: This will need to be removed in the future, as the upload promotion button
+          in the upload promotion page will invoke AmazonS3Service.uploadImage(image, promotionId) 
+          after an promotion has been successfully saved to our backend db
+      */}
```

Comment:
Good point, could we track this TODO in a ticket and link the ticket to this comment?

Reply:
yes! I've created PP-75 and PP-76 to track these [https://promopal.atlassian.net/browse/PP-595](url)

Reply:
Awesome thanks!